client/src/main/java/io/pravega/client/segment/impl/SegmentInputStreamFactoryImpl.java

```diff
@@ -25,30 +25,55 @@
 
     private final Controller controller;
     private final ConnectionFactory cf;
-    
+
     @Override
-    public SegmentInputStream createInputStreamForSegment(Segment segment) {
-        return createInputStreamForSegment(segment, SegmentInputStreamImpl.DEFAULT_BUFFER_SIZE);
+    public EventSegmentInputStream createEventInputStreamForSegment(Segment segment) {
+        return createEventInputStreamForSegment(segment, SegmentInputStreamImpl.DEFAULT_BUFFER_SIZE);
     }
 
     @Override
-    public SegmentInputStream createInputStreamForSegment(Segment segment, long endOffset) {
-        return getSegmentInputStream(segment, endOffset, SegmentInputStreamImpl.DEFAULT_BUFFER_SIZE);
+    public EventSegmentInputStream createEventInputStreamForSegment(Segment segment, long endOffset) {
+        return getEventSegmentInputStream(segment, endOffset, SegmentInputStreamImpl.DEFAULT_BUFFER_SIZE);
     }
 
     @Override
-    public SegmentInputStream createInputStreamForSegment(Segment segment, int bufferSize) {
-        return getSegmentInputStream(segment, Long.MAX_VALUE, bufferSize);
+    public EventSegmentInputStream createEventInputStreamForSegment(Segment segment, int bufferSize) {
+        return getEventSegmentInputStream(segment, Long.MAX_VALUE, bufferSize);
+    }
+
+    private EventSegmentInputStream getEventSegmentInputStream(Segment segment, long endOffset, int bufferSize) {
+        String delegationToken = Futures.getAndHandleExceptions(controller.getOrRefreshDelegationTokenFor(segment.getScope(),
+                                                                                                          segment.getStream()
+                                                                                                                 .getStreamName()),
+                                                                RuntimeException::new);
+        AsyncSegmentInputStreamImpl async = new AsyncSegmentInputStreamImpl(controller, cf, segment, delegationToken);
+        try {
+            Exceptions.handleInterrupted(() -> async.getConnection().get());
+        } catch (ExecutionException e) {
+            log.warn("Initial connection attempt failure. Suppressing.", e);
```

Comment:
I don't mean to imply the exception type. It's a failure to make an initial connection. getConnection() is creating the connection. I am suppressing it because this getConnection() call will end up being the first call the user makes but it takes a network round trip to establish. This was attempting to start that process earlier to lower latency in case the user needed to do something else first. 

But I realized based on your comment: There is no reason to wait on this future at all. I can just call getConnection() and return ignoring the result.